inc/c_util/two_d_array_ll.h

```diff
@@ -15,10 +15,236 @@
+    T* result;                                                                                                                                             \
+    /* Codes_SRS_TWO_D_ARRAY_07_019: [ If two_d_array is NULL, TWO_D_ARRAY_GET(T) shall fail return NULL. ]*/                                              \
+    if (two_d_array == NULL)                                                                                                                               \
+    {                                                                                                                                                      \
+        LogError("Invalid arguments: TWO_D_ARRAY_HANDLE two_d_array=%p", two_d_array);                                                                     \
+    }                                                                                                                                                      \
+    else                                                                                                                                                   \
+    {                                                                                                                                                      \
+        /* Codes_SRS_TWO_D_ARRAY_07_020: [ If row_index is equal or greater than row_size, TWO_D_ARRAY_GET(T) shall fail return NULL. ]*/                  \
+        if (row_index >= two_d_array->rows)                                                                                                                \
+        {                                                                                                                                                  \
+            LogError("Invalid arguments: uint64_t row_index=%" PRIu64 " out of bound, total_rows = %" PRIu64, row_index, two_d_array->rows);              \
+        }                                                                                                                                                  \
+        else                                                                                                                                               \
+        {                                                                                                                                                  \
+            result = two_d_array->row_arrays[row_index];                                                                                                   \
+            if(result == NULL)                                                                                                                             \
+            {                                                                                                                                              \
+            /* Codes_SRS_TWO_D_ARRAY_07_021: [ If the array stored in row_index is NULL, TWO_D_ARRAY_GET(T) shall fail and return NULL. ]*/                \
```

Comment:
>/*  [](http://example.com/codeflow?start=12&length=3)

Nitpick: add a tab here to indent the Codes like the line of code.

Reply:
fixed